api4/plugin_test.go

```diff
@@ -927,6 +929,105 @@ func TestGetLocalPluginInMarketplace(t *testing.T) {
 	})
 }
 
+func TestGetPrepackagedPluginInMarketplace(t *testing.T) {
+	th := Setup().InitBasic()
+	defer th.TearDown()
+
+	samplePlugins := []*model.MarketplacePlugin{
+		{
+			BaseMarketplacePlugin: &model.BaseMarketplacePlugin{
+				HomepageURL: "https://example.com/mattermost/mattermost-plugin-nps",
+				IconData:    "https://example.com/icon.svg",
+				DownloadURL: "www.github.com/example",
+				Manifest: &model.Manifest{
+					Id:               "marketplace.test",
+					Name:             "marketplacetest",
+					Description:      "a marketplace plugin",
+					Version:          "0.1.2",
+					MinServerVersion: "",
+				},
+			},
+			InstalledVersion: "",
+		},
+	}
+
+	testServer := httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		res.WriteHeader(http.StatusOK)
+		json, err := json.Marshal([]*model.MarketplacePlugin{samplePlugins[0]})
+		require.NoError(t, err)
+		res.Write(json)
+	}))
+	defer testServer.Close()
+
+	th.App.UpdateConfig(func(cfg *model.Config) {
+		*cfg.PluginSettings.Enable = true
+		*cfg.PluginSettings.EnableMarketplace = true
+		*cfg.PluginSettings.MarketplaceUrl = testServer.URL
+	})
+
+	t.Run("get remote and prepackaged plugins", func(t *testing.T) {
+		th.App.UpdateConfig(func(cfg *model.Config) {
+			*cfg.PluginSettings.EnableRemoteMarketplace = true
+			*cfg.PluginSettings.EnableUploads = true
+		})
+
+		manifest := &model.Manifest{
+			Version: "0.0.1",
+			Id:      "prepackaged.test",
+		}
+		newerPrepackagePlugin := &plugin.PrepackagedPlugin{
+			Manifest: manifest,
+		}
+
+		env := th.App.GetPluginsEnvironment()
+		env.SetPrepackagedPlugins([]*plugin.PrepackagedPlugin{newerPrepackagePlugin})
+
+		plugins, resp := th.SystemAdminClient.GetMarketplacePlugins(&model.MarketplacePluginFilter{})
+		CheckNoError(t, resp)
+
+		require.Len(t, plugins, 2)
```

Comment:
can we assert on the plugin ids, or the manifest maybe, that would confirm if the result is expected.